### ISSUE TITLE
Torch Migration

### ISSUE THREAD

**Issue by csolorio** [Microsoft/CNTK#2526]
I left open a previous case where I was asking some torch->cntk migration issues, but I think it got lost. Let me ask in a more 'generic' thread:

During my quest to make the migration of a model, I encountered the following cases in torch that I'm not sure I can reimplement in CNTK: 
1. Dynamically changing the model during training.
```
        if isSwap then
            --Fast model swap
            tempModel = self.model
            self.model = self.util:swapModel(self.model, scaleIdx)
        end

        self.currentErr = 0
	self.model:forward(self.input.train)
	self.currentErr = self.currentErr + self.criterion(self.model.output, self.target)
	self.model:backward(self.input.train, self.criterion.gradInput)

        if isSwap then
            --Return to original model
            self.model = tempModel
        end
```
where ``swapModel`` chooses some branchs from parallel containers and concat tables, like:
```
	local sModel = nn.Sequential()
	sModel
		:add(model:get(1))
		:add(model:get(2))
		:add(model:get(3):get(index))
		:add(model:get(4))
		:add(model:get(5):get(index))
		:add(model:get(6):get(index))
```
Can I use the cntk functions of the same name (forward/backward)? Incidentally, does CNTK have some mechanism to add or remove layers from a whole network after creation? I think I did see something like that in the documentation but I haven't been able to find it again :(  

2. Skipping an update to the nn state if some condition is false, like in:
``` 
        if self.currentErr < someThreshold 
            -- Equivalent to use the specified learner and update state
            self.optim(self.feval, self.params, self.optimState)
        else
            print(('Warning: Error is too large! Skip this batch. (Err: %.6f)'):format(self.currentErr))
            torch.save('../../skipBatch.t7', {self.input.train, self.target})
        end
```

3. I don't see a  L1 loss function in the docs, can it be implemented simply by using a custom loss function like (abs(a-b))/size(dim)?

4. Are there equivalents to Concat and Parallel containers? 
If not, could a concat table be created by making several layers share the same inputs and somehow join the output tensors? What could I do for the parallel container case?

Thanks in advance and keep doing a great work!
